src/main/java/VASSAL/build/module/Chatter.java

```diff
@@ -115,55 +134,245 @@ public void actionPerformed(ActionEvent e) {
         input.setText(""); //$NON-NLS-1$
       }
     });
-    input.setMaximumSize(new Dimension(input.getMaximumSize().width,
-                                       input.getPreferredSize().height));
+    input.setMaximumSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height));
+    
+    FontMetrics fm = getFontMetrics(myFont);
+    int fontHeight = fm.getHeight();
+
+    conversation.setPreferredSize(new Dimension(input.getMaximumSize().width, fontHeight * 10));
+
     scroll.setViewportView(conversation);
+    scroll.getVerticalScrollBar().setUnitIncrement(input.getPreferredSize().height); //Scroll this faster
     add(scroll);
     add(input);
+    
+    setPreferredSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height + conversation.getPreferredSize().height));
   }
 
   private String formatChat(String text) {
     final String id = GlobalOptions.getInstance().getPlayerId();
-    return "<" + (id.length() == 0 ? "("+getAnonymousUserName()+")" : id) + "> - " + text; //$NON-NLS-1$ //$NON-NLS-2$
+    return String.format("&lt;%s&gt; - %s", id.isEmpty() ? "(" + getAnonymousUserName() + ")" : id, text); //HTML-friendly angle brackets //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
   }
 
   public JTextField getInputField() {
     return input;
   }
 
+  // To make the player colors easy to override in a custom class 
+  // (my modules have logic to assign individual player colors -- beyond the scope of the present effort but a perhaps a fun future addition)
+  protected String getChatStyle(String s) {   
+    if (s.startsWith(formatChat("").trim())) { //$NON-NLS-1$
+      return "mychat";      
+    } 
+    else {
+      return "other";
+    }   
+  }
```

Comment:
I'd write this as a ternary: `return s.startsWith(formatChat("").trim()) ? "mychat" : "other";`

Reply:
Fixed, checked in.